File: OLSource1/v1/vs140/codesnippet/VisualBasic/how-to--extend-a-query-by-using-code_2.vb
        Private Sub CustomersWhoBoughtProduct_PreprocessQuery _
            (ByVal ProductID As System.Nullable(Of Short), _
             ByRef query As System.Linq.IQueryable(Of LS_Queries_VB.Customer))
            query = From myCustomers In query
                From myOrders In myCustomers.Orders
                From myOrderDetails In myOrders.Order_Details
                Where myOrderDetails.Product.ProductID = ProductID
                Select Customers
        End Sub

File: OLSource1/v1/VS_csharp/codesnippet/CSharp/how-to--extend-a-query-by-using-code_2.cs
        partial void CustomersWhoBoughtProduct_PreprocessQuery
            (short? ProductID, ref IQueryable<Customer> query)
        {
            query = from myCustomers in query
                    from myOrders in myCustomers.Orders
                    from myOrderDetails in myOrders.Order_Details
                    where myOrderDetails.Product.ProductID == ProductID
                    select myCustomers;
        }